lagranlucha-client/Assets/_Main/Scripts/Game/UI/OrderUI.cs

```diff
@@ -0,0 +1,150 @@
+﻿using UnityEngine;
+
+using DG.Tweening;
+using Zenject;
+using System;
+
+namespace LaGranLucha.Game
+{
+    public class OrderUI : MonoBehaviour
+    {
+        #region FIELDS
+
+        private const float TrayMovementDuration = 0.5f;
+        private const float Delay = 1f;
+
+        [Inject] private OrderManager orderManager = null;
+        [Inject] private GameManager gameManager = null;
+
+        [SerializeField] private Transform withHamburguerContainer = null;
+        [SerializeField] private Transform noHamburguerContainer = null;
+        [SerializeField] private Transform onlyOneContainer = null;
+        [SerializeField] private GameIngredient gameIngredientPrefab = null;
+
+        private Transform hamburguerContainer = null;
+        private Transform friesContainer = null;
+        private Transform drinkContainer = null;
+        private RectTransform rectTransform = null;
+        private Vector2 startPosition = Vector2.zero;
+
+        #endregion
+
+        #region EVENTS
+
+        public event Action onTrayCleaned;
+
+        #endregion
+
+        #region BEHAVIORS
+
+        private void Awake()
+        {
+            rectTransform = GetComponent<RectTransform>();
+            startPosition = rectTransform.anchoredPosition;
+            orderManager.onHamburguerIngredientSubmit += SendHamburguerIngredient;
+            orderManager.onFriesIngredientSubmit += SendFriesIngredient;
+            orderManager.onDrinkIngredientSubmit += SendDrinkIngredient;
+            orderManager.onOrderCompleted += SendTrayToSide;
+            orderManager.onHamburguerReset += Resethamburguer;
+            orderManager.onNewOrderStart += OrderStart;
+            gameManager.onRoundStart += StartRound;
+        }
+
+        private void OnDestroy()
+        {
+            orderManager.onHamburguerIngredientSubmit -= SendHamburguerIngredient;
+            orderManager.onFriesIngredientSubmit -= SendFriesIngredient;
+            orderManager.onDrinkIngredientSubmit -= SendDrinkIngredient;
+            orderManager.onOrderCompleted -= SendTrayToSide;
+            orderManager.onHamburguerReset -= Resethamburguer;
+            orderManager.onNewOrderStart -= OrderStart;
+            gameManager.onRoundStart -= StartRound;
+        }
+
+        private void StartRound(int round)
+        {
+            BringTrayToCenter();
+        }
+
+        private void OrderStart(Order order)
+        {
+            if (order.HaveHamburger && (order.HaveDrink || order.HaveFries))
+            {
+                friesContainer = withHamburguerContainer.GetChild(0);
+                drinkContainer = withHamburguerContainer.GetChild(1);
```

Comment:
These numbers look a little bit magical, what do they stand for?

Reply:
Everything  container have its part for adding hamburguer or fries or drinks I make constants to be more evident